### ISSUE TITLE
`app.html` doesn't respect base path setting

### ISSUE THREAD

**Issue by andrewbanchich** [sveltejs/kit#2852]
### Describe the bug

The skeleton template creates an `app.html` file that contains:

```html
<!DOCTYPE html>
<html lang="en">
	<head>
		<meta charset="utf-8" />
		<link rel="icon" href="/favicon.png" />
		<meta name="viewport" content="width=device-width, initial-scale=1" />
		%svelte.head%
	</head>
	<body>
		<div id="svelte">%svelte.body%</div>
	</body>
</html>
```

Setting a base path using [this setting](https://kit.svelte.dev/docs#configuration-paths) breaks the favicon because it's hardcoded to use the root path (`https://mysite.com/favicon.png`), not the base (`https://mysite.com/foo/favicon.png`).

This is even more troublesome after adding a global stylesheet. The only alternative I've found is adding the stylesheet in `<svelte:head>` which is far from ideal.

I couldn't find documentation for `%svelte.head%` or `%svelte.body%`, but I think it would useful to have something like those that could access the config's base path value.

### Reproduction

https://github.com/sveltejs/kit

```
npm init svelte@next my-app
cd my-app
npm install
npm run dev -- --open
```

### System Info

```shell
System:
    OS: Linux 5.13 Pop!_OS 21.04
    CPU: (8) x64 11th Gen Intel(R) Core(TM) i5-1135G7 @ 2.40GHz
    Memory: 985.59 MB / 23.33 GB
    Container: Yes
    Shell: 3.3.0 - /usr/bin/fish
  Binaries:
    Node: 14.17.0 - /usr/local/bin/node
    Yarn: 1.22.17 - /usr/local/bin/yarn
    npm: 8.1.1 - /usr/local/bin/npm
  Browsers:
    Firefox: 94.0
```
```


### Severity

annoyance

### Additional Information

_No response_

**Comment by Alotor** [comment]
Have you find a workaround around this? I have the base path in the html but it's a bit messy to change it when deploying...

**Comment by andrewbanchich** [comment]
Other than the `<svelte:head>` one I mentioned above, no.

**Comment by andrewbanchich** [comment]
Is this something that should be addressed before 1.0 @benmccann ?

**Comment by benmccann** [comment]
Was this fixed by https://github.com/sveltejs/kit/pull/3234 and / or https://github.com/sveltejs/kit/pull/3346 ?

**Comment by andrewbanchich** [comment]
I upgraded SvelteKit to `1.0.0-next.232` to try this out again, but running `npm run build` no longer produces an `index.html` in the build directory if I set `kit.paths.base`. It doesn't throw any errors either, so I'm not sure why.

Setting `kit.paths.assets` requires it to be an absolute URL, so I'd prefer to use `base` either way to avoid having to manually build for separate domains.

**Comment by nhe23** [comment]
Meanwhile the generated app.html uses:
```html
<link rel="icon" href="%svelte.assets%/favicon.png" />
```
which initially resolved the issue.
However there was a bug that messed up prerendering when base path was configured. This has been resolved by #3500. So now everything works as expected and this issue can probably be closed.
